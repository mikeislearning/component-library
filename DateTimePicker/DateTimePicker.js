import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import { PropTypes } from 'prop-types';
import DatePicker from 'react-native-modal-datetime-picker';

import { FormError } from 'app/components';

import { momentDate } from 'app/utils';

import styles from './DateTimePicker.styles';

const formatDate = (date) => {
  return momentDate(date, 'MMM DD, YYYY');
};

export default class DateTimePicker extends React.Component {
  static propTypes = {
    onConfirm: PropTypes.func,
    value: PropTypes.string,
    errorMsg: PropTypes.string,
    errors: PropTypes.bool,
  };

  state = {
    isDateTimePickerVisible: false,
    date: this.props.value ? formatDate(this.props.value) : '',
  };

  showDateTimePicker = () => {
    this.setState({
      isDateTimePickerVisible: true,
    });
  };

  hideDateTimePicker = () => {
    this.setState({
      isDateTimePickerVisible: false,
    });
  };

  handleDatePicked = (date) => {
    const formattedDate = formatDate(date);
    this.setState({
      date: formattedDate,
    });
    this.hideDateTimePicker();
    this.props.onConfirm(formattedDate);
  };

  render() {
    const { date, isDateTimePickerVisible } = this.state;
    const { errorMsg, errors } = this.props;
    const initDate = date ? new Date(date) : new Date();
    const dateTimeContainerStyles = [styles.dateTimeContainer];
    const dateTimeTextStyles = [styles.dateTimeText];
    if (!date) {
      dateTimeContainerStyles.push(styles.dateTimeContainerEmpty);
      dateTimeTextStyles.push(styles.dateTimeTextEmpty);
    }
    if (errors) dateTimeContainerStyles.push(styles.errorInput);

    return (
      <View style={styles.container}>
        <TouchableOpacity
          activeOpacity={1}
          onPress={this.showDateTimePicker}
          style={dateTimeContainerStyles}
        >
          <Text style={dateTimeTextStyles}>{date || 'Select a date'}</Text>
        </TouchableOpacity>
        <DatePicker
          date={initDate}
          isVisible={isDateTimePickerVisible}
          onConfirm={this.handleDatePicked}
          onCancel={this.hideDateTimePicker}
        />
        {errorMsg && <FormError text={errorMsg} />}
      </View>
    );
  }
}
