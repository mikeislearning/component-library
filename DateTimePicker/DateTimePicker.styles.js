import { StyleSheet } from 'react-native';
import { colors, fonts } from 'app/theme';

const styles = StyleSheet.create({
  container: {
    marginBottom: 10,
  },
  dateTimeContainer: {
    backgroundColor: colors.white,
    borderColor: colors.black,
    borderRadius: 8,
    borderWidth: 1,
    height: 50,
    justifyContent: 'center',
    paddingHorizontal: 5,
  },
  dateTimeContainerEmpty: {
    backgroundColor: colors.gray.light,
    borderColor: colors.gray.medium,
  },
  dateTimeText: {
    color: colors.black,
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: 16,
  },
  dateTimeTextEmpty: {
    color: colors.gray.medium,
    fontFamily: fonts.family.messinaSans.lightItalic,
  },
  errorInput: {
    borderColor: colors.red.light,
  },
  placeholder: {
    color: colors.gray.medium,
  },
});

export default styles;
