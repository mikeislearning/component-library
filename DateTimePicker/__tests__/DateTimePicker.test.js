import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import DateTimePicker from '../DateTimePicker';

describe('<Dropdown />', () => {
  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<DateTimePicker value="2019-09-03" />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
