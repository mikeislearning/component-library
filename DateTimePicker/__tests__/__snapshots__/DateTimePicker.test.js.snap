// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> renders correctly 1`] = `
<React.Fragment>
  <TouchableOpacity
    activeOpacity={1}
    onPress={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#000000",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 10,
          "minWidth": 280,
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "MessinaSans-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      Sep 03, 2019
    </Text>
  </TouchableOpacity>
  <CustomDatePickerIOS
    cancelTextIOS="Cancel"
    confirmTextIOS="Confirm"
    date={2019-09-03T04:00:00.000Z}
    dismissOnBackdropPressIOS={true}
    hideTitleContainerIOS={false}
    isVisible={false}
    mode="date"
    neverDisableConfirmIOS={false}
    onCancel={[Function]}
    onConfirm={[Function]}
    onDateChange={[Function]}
    onHideAfterConfirm={[Function]}
    reactNativeModalPropsIOS={Object {}}
    titleIOS="Pick a date"
  />
</React.Fragment>
`;
